### PR TITLE
Everywhere: Make the codebase more architecture aware

### PR DESCRIPTION
**Pull request by Undefine-Unity**
Mainly not assuming that not X86_64 is I686 and in reverse.

### CODE REVIEW

**Review by linusg — CHANGES_REQUESTED** [review]
I'm very much in favour of explicit arch macro branches and banning `#else` for this purpose, but not like this - pick one of `#error Unknown architecture`, `TODO()` or stub code, not all of them. Preferably the first.

File: Userland/Utilities/functrace.cpp [hunk 1]
```diff
@@ -58,14 +58,17 @@ static void print_syscall(PtraceRegisters& regs, size_t depth)
...
     outln("=> {}SC_{}({:#x}, {:#x}, {:#x}){}",
         begin_color,
         Syscall::to_string((Syscall::Function)regs.rax),
         regs.rdx,
         regs.rcx,
         regs.rbx,
         end_color);
+#else
+#    error Unknown architecture
+
```
Comment (bgianfo) [inline Userland/Utilities/functrace.cpp:71]: Extra newline here?

File: Userland/Libraries/LibC/string.cpp [hunk 2]
```diff
@@ -164,6 +168,8 @@ void* memset(void* dest_ptr, int c, size_t n)
         : "memory");
     return dest_ptr;
 }
+#elif !ARCH(X86_64)
```
Comment (bgianfo) [inline Userland/Libraries/LibC/string.cpp:171]: nit: Please make this a normal if/else check and then move the comment about the x86_64 assembler implementation into the `ARCH(X86_64)` check, and add a standard `#else` + Unknown architecture message.

**Review by bgianfo — APPROVED** [review]
LGTM!

**Pull request merged**